nodeadm/cmd/nodeadm/init/init.go

```diff
@@ -111,21 +122,76 @@ func (c *initCmd) Run(log *zap.Logger, opts *cli.GlobalOptions) error {
 
 // Various initializations and verifications of the NodeConfig and
 // perform in-place updates when allowed by the user
-func enrichConfig(cfg *api.NodeConfig) error {
-	zap.L().Info("Fetching instance details..")
-	instanceDetails, err := util.FetchInstanceDetails()
+func enrichConfig(log *zap.Logger, cfg *api.NodeConfig) error {
+	log.Info("Fetching instance details..")
+	instanceDetails, err := api.GetIMDSInstanceDetails(context.TODO(), imds.New(imds.Options{}))
 	if err != nil {
 		return err
 	}
 	cfg.Status.Instance = *instanceDetails
-	zap.L().Info("Instance details populated", zap.Reflect("details", instanceDetails))
+	log.Info("Instance details populated", zap.Reflect("details", instanceDetails))
 
 	if featuregates.DefaultFalse(featuregates.DescribeClusterDetails, cfg.Spec.FeatureGates) {
-		zap.L().Info("Populating cluster details using a describe-cluster call..")
-		if err := featuregates.PopulateClusterDetails(cfg.Spec.Cluster.Name, cfg); err != nil {
+		log.Info("Populating cluster details using a describe-cluster call..")
+		// use instance region since cluster region isn't guarenteed to exist
+		awsConfig := aws.NewConfig().WithRegion(instanceDetails.Region)
+		eksClient := eks.New(session.Must(session.NewSession(awsConfig)))
+		if err := populateClusterDetails(eksClient, cfg.Spec.Cluster.Name, cfg); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
+
+// Discovers all cluster details using a describe call to the eks endpoint and
+// updates the value of the config's `ClusterDetails` in-place
+func populateClusterDetails(eksClient *eks.EKS, clusterName string, cfg *api.NodeConfig) error {
```

Comment:
since i donn't theres another eks sdk consumer at the moment, im just pushing this as close to main as possible :)

Reply:
LGTM